### PR TITLE
Convert `historyItemsStore` to ts+composition

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by davelopez** [comment]
Since this is a bug fix shouldn't it target 23.1?

**Comment by ahmedhamidawan** [comment]
> Since this is a bug fix shouldn't it target 23.1?

Good point, my bad, change the target branch

File: client/src/stores/historyItemsStore.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/**
+ * Requests history items by reacting to changes of filter props passed
+ * to the history items provider used in the history panel.
+ */
```
Comment (ElectronicBlueberry) [inline client/src/stores/historyItemsStore.ts]: I feel like this comment is either outdated, or unclear: I fail to find the mentioned provider and it's connection to this store

Comment (ahmedhamidawan) [reply client/src/stores/historyItemsStore.ts]: Yes very good point, we don't use the provider and instead watch changes in variables in `HistoryPanel`
Will update that

File: client/src/components/History/CurrentHistory/HistoryPanel.vue [hunk 2]
```diff
@@ -416,7 +411,7 @@ export default {
             if (data) {
                 const dataSource = data.history_content_type === "dataset" ? "hda" : "hdca";
                 if (data.history_id != this.historyId) {
-                    copyDataset(data.id, this.historyId, data.history_content_type, dataSource)
+                    await copyDataset(data.id, this.historyId, data.history_content_type, dataSource)
                         .then(() => {
                             if (data.history_content_type === "dataset") {
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryPanel.vue]: If you use `await` you can replace the `then/catch` with a try/catch and increase a bit readability while reducing nesting.

File: client/src/stores/historyItemsStore.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+const limit = 100;
+const queue = new LastQueue();
+
+export const useHistoryItemsStore = defineStore("historyItemsStore", () => {
+    const items: Ref<Record<string, HistoryItem[]>> = ref({});
+    const itemKey = ref("hid");
+    const totalMatchesCount = ref(undefined);
+    const lastCheckedTime = ref(new Date());
+    const lastUpdateTime = ref(new Date());
+    const relatedItems: Ref<Record<string, boolean>> = ref({});
```
Comment (davelopez) [inline client/src/stores/historyItemsStore.ts]: If I recall correctly, you don't need to import and use `Ref` to define the type, you can simply use:
```suggestion
    const relatedItems = ref<Record<string, boolean>>({});
```

Comment (ahmedhamidawan) [reply client/src/stores/historyItemsStore.ts]: Yes, I have used the `Ref` type definition in other places as well, realizing that maybe should've just done this instead, will make that change here though

**Pull request merged**